Frontend changes

Cxmrykk opened the pull request:
**Week 5-6 | 08/09/2024**

- Includes changes from Chris's branch up to 31c0c8bca63ad0bb13d41feff03e1d1fa40b120c

Cxmrykk committed:
add camera/gallery buttons

Cxmrykk committed:
Implement history page

Cxmrykk committed:
change album button behaviour

Cxmrykk committed:
existing home -> history

Cxmrykk committed:
use asyncstorage for history items

Cxmrykk committed:
add mitt

Cxmrykk committed:
fix overlay not opening on first tap

oSirhcel commented on frontend/app/(tabs)/history.tsx:
```diff
@@ -0,0 +1,267 @@
+      loadScannedItems();
+    };
+
+    scannedItemsEmitter.on('newItemScanned', updateScannedItemsListener);
```
I think using an emitter is a bit overkill, because the user will never be on both the history page and the scanner page at the same time.

A better approach would be to fetch the saved items when the user enters the history page from the local storage (ideally, we would cache it so that it would only fetch when changes are made).

oSirhcel commented on frontend/app/(tabs)/scan.tsx:
```diff
@@ -0,0 +1,314 @@
+  saveToLibraryAsync,
+  getAssetsAsync,
+} from 'expo-media-library';
+import * as ImagePicker from 'expo-image-picker';
```
expo-image-picker isn't in the package.json file

oSirhcel reviewed (changes requested):
I'm getting an error on my phone where the app crashes when the bottom drawer is open, think it has something to do with when the newScannedItem is saved.

I really like the visual changes made the bottom drawer and the structural changes of the app, but we'll probably have to change how we approach storage handling and fetching as its currently beyond our MVP unless we want to expand our scope.

For now, I think we should remove/stash the emitter, and mitt as a dependency, remove everything involving saving and then merge to the dev branch (just so we can have some progress in dev) and then feel free to explore the options in your own branch.

Good work overall 👍

Cxmrykk committed:
Remove storage handling, fixes

Cxmrykk commented:
- Removed existing storage handling and event emitter
- Replaced the history page with mock data

Also that bug should be fixed, although needs to be verified on IOS (I am testing on Android)
